e2e/test/Helpers/templates/FaultInjectionPoolingOverAmqp.cs

```diff
@@ -175,12 +175,12 @@ public static class FaultInjectionPoolingOverAmqp
                     {
                         if (FaultInjection.FaultShouldDisconnect(faultType))
                         {
-                            // 4 is the minimum notification count: connect, fault, reconnect, disable.
-                            Assert.IsTrue(amqpConnectionStatuses[i].ConnectionStatusChangeCount >= 4, $"The expected connection status change count for {testDevices[i].Id} should equals or greater than 4 but was {amqpConnectionStatuses[i].ConnectionStatusChangeCount}");
+                            // Notifications: connect, fault, reconnect, disable.
```

Comment:
I am not sure why we were checking >= earlier. I think we should be checking for an exact number here. Let me know if there was a reason to have >=.

Reply:
The reason we are checking >= count is because the device could get faulted due to some network instability etc (prior to any fault being injected). This ensures that the min no of connection change events take place.

This change needs to be reverted as well.

Reply:
Thanks for this clarification.

Reply:
By that logic, do we need >= 2 in the check below then? Is that the reason for failure and not really device uniqueness?

Reply:
It's possible; there could always be some amount of instability due to network disconnects, and I remember we had a retry logic within these tests (at some point). I am not sure if we still have that logic, but personally, I would have preferred to have these tests retry, rather than assert with a >= check.